### PR TITLE
Project upgrade docs

### PR DESCRIPTION
**Pull request by nmashton**
Informed by my experience of upgrading Cotton University to use our project template, here are some docs covering what that entails.

### CODE REVIEW

File: docs/dev/upgrading.rst [hunk 1]
```diff
@@ -0,0 +1,188 @@
+Upgrading to the project template
```
Comment (jbradberry) [inline docs/dev/upgrading.rst]: I'm not sure 'upgrading' is the right word.  "Converting legacy projects" perhaps?

**Comment by jbradberry** [comment]
You should note the importance of `conf/salt/top.sls` if you need to add custom roles (e.g. Solr)

**Comment by jbradberry** [comment]
Or even just adding optional states to an existing role (e.g. `forward_logs` for PaperTrail).

**Comment by jbradberry** [comment]
Perhaps note `conf/pillar/<environment>.sls` as the place where environment deployment configuration and secrets go.

File: docs/dev/upgrading.rst [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+Files and requirements
+~~~~~~~~~~~~~~~~~~~~~~
+
+Copy over the following files from the DPT base:
+
+-  ``fabfile.py``: the Fabric script used to automate provisioning and
+   deployment
+-  ``install_salt.sh``: used by Fabric when setting up the Salt master
+   during provisioning
+-  ``Makefile``: used for generating secrets
```
Comment (jbradberry) [inline docs/dev/upgrading.rst]: ... and so much more!

**Comment by jbradberry** [comment]
`fabfile.py` needs to be updated (or did we do this the recommended way?) with the addresses of the environments.

**Comment by jbradberry** [comment]
`conf/pillar/devs.sls` to set up the dev's ssh keys in the environments.

**Comment by jbradberry** [comment]
There's probably other stuff from https://github.com/caktus/cottonu/pull/9 and https://github.com/caktus/cottonu/pull/12 that could be noted.

File: docs/dev/upgrading.rst (hunk 2 above)
Comment (nmashton) [inline docs/dev/upgrading.rst]: Yes! But in this context, this property is what we care about.

Comment (jbradberry) [inline docs/dev/upgrading.rst]: Perhaps something more imperative, extracted out from the 'list of files' format here.  "You will need to do `make foobar` to generate the new Django `SECRET_KEY` value."

**Comment by jbradberry** [comment]
I'm more than a little bit annoyed by the CI failure (see #274), but in general I think this looks good.  :shipit:

**Comment by dpoirier** [comment]
@nmashton any reason not to merge this?  It looks like jrb approved it.

**Comment by jbradberry** [comment]
@dpoirier I can think of no reason not to, and plenty to do so.

**Comment by vkurup** [comment]
I made a few minor updates, but this looks good to me. Merging...

**Pull request merged**